tests/libblight_protocol/test.c

```diff
@@ -0,0 +1,537 @@
+    int fds[2];
+    assert(
+        socketpair(AF_UNIX, SOCK_STREAM | SOCK_NONBLOCK, 0, fds) != -1
+        && "Failed to create socket pair"
+    );
+    int clientFd = fds[0];
+    int serverFd = fds[1];
+    blight_data_t data = malloc(10);
+    assert(data != NULL && "Malloc failed");
+    assert(blight_send_blocking(serverFd, data, 10) == 0);
+    blight_data_t data2 = NULL;
+    assert(blight_recv(clientFd, &data2, 10) == 0);
+    assert(data2 != NULL);
+    assert(memcmp(data, data2, 10) == 0);
+    free(data);
+    free(data2);
+    close(clientFd);
+    close(serverFd);
+}
+void test_blight_surface_to_fbg(int fd, blight_surface_id_t _identifier, blight_buf_t* _buf){
+    struct _fbg* fbg = blight_surface_to_fbg(fd, _identifier, _buf);
+    assert(fbg == NULL);
+    assert(errno == EINVAL);
+    blight_buf_t* buf = blight_create_buffer(10, 10, 100, 100, 100 * 3, Format_RGB32);
+    assert(buf != NULL);
+    blight_surface_id_t identifier = blight_add_surface(bus, buf);
+    assert(identifier > 0);
+    fbg = blight_surface_to_fbg(fd, identifier, buf);
+    assert(fbg != NULL);
+    assert(fbg->size == (int)(buf->height * buf->stride));
+    fbg_clear(fbg, 0);
+    fbg_rect(fbg, fbg->width / 2 - 32, fbg->height / 2 - 32, 16, 16, 0, 255, 0);
+    fbg_draw(fbg);
+    fbg_flip(fbg);
+    fbg_close(fbg);
+}
+void test_blight_move_surface(int fd){
+    blight_buf_t* buf = blight_create_buffer(10, 10, 100, 100, 100 * 3, Format_RGB32);
+    assert(buf != NULL);
+    blight_surface_id_t identifier = blight_add_surface(bus, buf);
+    assert(identifier > 0);
+    struct _fbg* fbg = blight_surface_to_fbg(fd, identifier, buf);
+    assert(fbg != NULL);
+    assert(fbg->size == (int)(buf->height * buf->stride));
+    fbg_clear(fbg, 0);
+    fbg_rect(fbg, fbg->width / 2 - 32, fbg->height / 2 - 32, 16, 16, 0, 255, 0);
+    fbg_draw(fbg);
+    fbg_flip(fbg);
+    int res = blight_move_surface(fd, identifier, buf, 200, 200);
+    assert(res >= 0);
+    assert(buf->x == 200);
+    assert(buf->y == 200);
+    fbg_draw(fbg);
+    fbg_flip(fbg);
+    // TODO - Figure out why box doesn't move, even though move event happened in display server logs
+    fbg_close(fbg);
+    res = blight_remove_surface(fd, identifier);
+    assert(res == 0);
+}
+void test_blight_thread(int fd, blight_surface_id_t identifier){
+    struct blight_thread_t* thread = blight_start_connection_thread(fd);
+    blight_data_t response = NULL;
+    int res = blight_send_message(fd, Ping, 1, 0, NULL, 0, &response);
+    assert(res == 0);
+    assert(response == NULL);
+    res = blight_send_message(fd, List, 2, 0, NULL, 0, &response);
+    // We have one surface right now, the one from test_blight_add_surface
+    assert(res == sizeof(blight_surface_id_t));
+    assert(response != NULL);
+    free(response);
+    blight_surface_id_t id = *((blight_surface_id_t*)response);
+    fprintf(stderr, "%d == %d\n", id, identifier);
+    assert(id == identifier);
```

Comment:
@coderabbitai Why would this never be true? The data is being populated from a `std::vector::data()` call before being sent down the pipe, and we expect there to be only one value in it, and the value of `identifier` would be the value being sent.